### DISCUSSION TITLE
Inconsistent Order of Instructions in Phantom Wallet Transactions on Chrome

### DISCUSSION THREAD

**Discussion by ninjatech-rahul** [phantom/docs#91] [Help & Support]
### Summary

I'm invoking a POST API that sends a transaction to the smart contract. To ensure the durability of the transaction, I'm using a nonce account to ensure its durability.
I've create a new transaction object and added nonce advance instruction and the main instruction object of the SC to it, and then signed the transaction.

On Firefox, everything is working fine, and I'm using Phantom Wallet version '23.5.0.1'.

However, on Chrome, here Phantom Wallet version '23.12.0',  when I invoke the same API, Phantom Wallet sends a warning that the 'transaction expired'. When I investigate I find out that this is due to the ordering of the instructions. 
On Firefox the order of the instructions in the transaction object is [nonce account, main instruction, system instructions].
While in Chrome the order changed to  [system instructions, nonce account, main instruction]. 
Ultimately If I ignore warning and proceed anyway, the API crashes, indicating that the 'Blockhash not found' error.

I would like to understand how can I resolve the issue.

### Steps to Reproduce

1. I've initialized a nonce account.
2. Create the instruction which I want to send to the SC, via BE.
3. Create a transaction object and add nonce advance ins and SC's instruction object.
4. Sign the transaction object from wallet on Chrome, here a warning occurs- 'Transaction expired'
  4.1. When ignore warning, proceed anyway, it fails with Blockhash not found error.
  4.2. While debugging it is find out that instructions reordered on Chrome.

### Phantom Version

On Chrome - 23.12.0 

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment]
Hi @ninjatech-rahul,

Thank you for writing in!

Would you mind providing 2 example transactions that show this inconsistent instructions order, please?
